cmd/tsidp/tsidp.go

```diff
@@ -90,6 +104,57 @@ func main() {
 		if !anySuccess {
 			log.Fatalf("failed to listen on any of %v", st.TailscaleIPs)
 		}
+
+		// In order to support funneling out in local tailscaled mode, we need
+		// to add a serve config to forward the listeners we bound above and
+		// allow those forwarders to be funneled out.
+		sc, err := lc.GetServeConfig(ctx)
+		if err != nil {
+			log.Fatalf("could not get serve config: %v", err)
+		}
+		if sc == nil {
+			sc = new(ipn.ServeConfig)
+		}
+
+		// We watch the IPN bus just to get a session ID. The session expires
+		// when we stop watching the bus, and that auto-deletes the foreground
+		// serve/funnel configs we are creating below.
+		watcher, err = lc.WatchIPNBus(ctx, ipn.NotifyInitialState|ipn.NotifyNoPrivateKeys)
+		if err != nil {
+			log.Fatalf("could not set up ipn bus watcher: %v", err)
+		}
+		defer watcher.Close()
+		n, err := watcher.Next()
+		if err != nil {
+			log.Fatalf("could not get initial state from ipn bus watcher: %v", err)
+		}
+		if n.SessionID == "" {
+			log.Fatalf("missing sessionID in ipn.Notify")
+		}
+
+		// Create a foreground serve config that gets cleaned up when tsidp
+		// exits and the session ID associated with this config is invalidated.
+		foregroundSc := new(ipn.ServeConfig)
+		mak.Set(&sc.Foreground, n.SessionID, foregroundSc)
+		serverURL := strings.TrimSuffix(st.Self.DNSName, ".")
+		fmt.Printf("setting funnel for %s:%v\n", serverURL, uint16(*flagPort))
+
+		// tailscaled needs to be setting an HTTP header for funneled requests
+		// that older versions.
```

Comment:
Could you reword this comment, I'm not sure what it means. You might have dropped a word: `that older versions [require]` or similar?

Reply:
oops, my bad, fixed!